### PR TITLE
chore(experiment): Add docs for node service kill experiment

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <uditgaurav@gmail.com>

issue - https://github.com/litmuschaos/litmus/issues/1541

- Adding kubelet service kill chaos experiment docs

### CODE REVIEW

File: docs/kubelet-service-kill.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+<table>
+  <tr>
+    <th> Type </th>
+    <th> Description </th>
+    <th> Tested K8s Platform </th>
+  </tr>
+  <tr>
+    <td> Generic </td>
+    <td> Kill Kubelet Service of Application Node </td>
+    <td> GKE, EKS </td>
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: Can we make them same as csv file?

File: docs/kubelet-service-kill.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+</table>
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/#install-litmus)
+- Ensure that the `kubelet-service-kill` experiment resource is available in the cluster  by executing                         `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos/1.5.0?file=charts/generic/kubelet-service-kill/experiment.yaml)
+
+## Entry Criteria
+
+- Application Node is healthy before chaos injection
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: we are checking for the application status in pre/post chaos as well?

File: docs/kubelet-service-kill.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+## Exit Criteria
+
+- Application Node should be healthy post chaos injection
+
+## Details
+
+- This experiment Causes the application to become unreachable on account of node turning unschedulable (NotReady) due to kubelet service kill.
+- The kubelet service kill has been injected gracefully on a  node to make it unschedulable for a certain TOTAL_CHAOS_DURATION. The application node should be healthy once chaos is stopped and the services are reaccessable
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: The kubelet service has been stopped/killed on a node to make it unschedulable for a certain duration say `TOTAL_CHAOS_DURATION`. The application node should be healthy after the chaos injection and the services should be reaccessable

File: docs/kubelet-service-kill.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  </tr>
+   <tr>
+    <td> LIB  </td>
+    <td> The chaos lib used to inject the chaos </td>
+    <td> Optional </td>
+    <td> Defaults to `litmus` </td>
+  </tr>
+  <tr>
+    <td> RAMP_TIME </td>
+    <td> Period to wait before injection of chaos in sec </td>
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: Period to wait before & after injection of chaos in sec

**Review by ispeakc0de — CHANGES_REQUESTED** [review]

File: docs/kubelet-service-kill.md (hunk 2 above)
Comment (uditgaurav) [reply docs/kubelet-service-kill.md]: its application node

Comment (ispeakc0de) [reply docs/kubelet-service-kill.md]: This is also a entry criteria right?
https://github.com/litmuschaos/litmus/blob/master/experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml#L54

File: docs/kubelet-service-kill.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+- Application pods are healthy on the respective Nodes before chaos injection.
+
+## Exit Criteria
+
+- Application pods will be healthy post chaos injection.
+
+## Details
+
+- This experiment Causes the application to become unreachable on account of node turning unschedulable (NotReady) due to kubelet service kill.
+- The kubelet service has been stopped/killed on a node to make it unschedulable for a certain duration sai.e `TOTAL_CHAOS_DURATION`. The application node should be healthy after the chaos injection and the services should be reaccessable.
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: sai.e ?

File: docs/kubelet-service-kill.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+- Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/#install-litmus)
+- Ensure that the `kubelet-service-kill` experiment resource is available in the cluster  by executing                         `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos/1.5.0?file=charts/generic/kubelet-service-kill/experiment.yaml)
+
+## Entry Criteria
+
+- Application pods are healthy on the respective Nodes before chaos injection.
+
+## Exit Criteria
+
+- Application pods will be healthy post chaos injection.
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: Application pods and the node should be healthy post chaos injection

File: docs/kubelet-service-kill.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+</table>
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/#install-litmus)
+- Ensure that the `kubelet-service-kill` experiment resource is available in the cluster  by executing                         `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos/1.5.0?file=charts/generic/kubelet-service-kill/experiment.yaml)
+
+## Entry Criteria
+
+- Application pods are healthy on the respective Nodes before chaos injection.
```
Comment (ispeakc0de) [inline docs/kubelet-service-kill.md]: Application pods should be healthy before chaos injection

**Review by ispeakc0de — APPROVED** [review]

File: docs/kubelet-service-kill.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+
+#### Sample Rbac Manifest
+
+[embedmd]:# (https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/generic/kubelet-service-kill/rbac.yaml yaml)
+```yaml
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubelet-service-kill
```
Comment (imrajdas) [inline docs/kubelet-service-kill.md:66]: This service account name is not using in chaosengine spec

File: docs/kubelet-service-kill.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+</table>
+
+#### Sample ChaosEngine Manifest
+
+[embedmd]:# (https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/generic/kubelet-service-kill/engine.yaml yaml)
+```yaml
+apiVersion: litmuschaos.io/v1alpha1
+kind: ChaosEngine
+metadata:
+  name: nginx-chaos
```
Comment (imrajdas) [inline docs/kubelet-service-kill.md:151]: name should be kubelet-chaos. Change it in other places

File: docs/kubelet-service-kill.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+</table>
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/#install-litmus)
+- Ensure that the `kubelet-service-kill` experiment resource is available in the cluster  by executing                         `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos/1.5.0?file=charts/generic/kubelet-service-kill/experiment.yaml)
+
+## Entry Criteria
+
+- Application pods should be healthy before chaos injection.
```
Comment (imrajdas) [inline docs/kubelet-service-kill.md:30]: - Application pods should be healthy before chaos injection.
- Nodes should be healthy before chaos injection

File: docs/kubelet-service-kill.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+
+## Exit Criteria
+
+- Application pods and the node should be healthy post chaos injection.
```
Comment (imrajdas) [inline docs/kubelet-service-kill.md:34]: - Application pods should be healthy after chaos injection.
- Nodes should be healthy after chaos injection

**Review by imrajdas — CHANGES_REQUESTED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**